spark/src/main/scala/org/apache/spark/sql/delta/util/PartitionUtils.scala

```diff
@@ -117,7 +117,8 @@ object PartitionSpec {
 
 private[delta] object PartitionUtils {
 
-  val timestampPartitionPattern = "yyyy-MM-dd HH:mm:ss[.S]"
+  lazy val timestampPartitionPattern = "yyyy-MM-dd HH:mm:ss[.S]"
+  lazy val utcFormatter = TimestampFormatter("yyyy-MM-dd HH:mm:ss[.S]z", ZoneId.of("UTC"))
```

Comment:
Another aspect to consider:

Delta supports microsecond timestamps, and those can also be used for partitioning. Kernel stores in microseconds https://github.com/delta-io/delta/blob/573a57f62918d0cb8937ca8c9b4047f8d696cc9c/kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/PartitionUtils.java#L45 and the same in Iceberg itself since the partition tuple is always just a timestamp type microseconds since epoch.

Perhaps we should do the same here?